app/src/test/java/com/concordia/mcga/fragments/DirectionsTest.java

```diff
@@ -0,0 +1,348 @@
+        ArrayList<String> direction = myFragment.getCompleteDirectionsList();
+
+        assertTrue(direction.get(index).equals("Up"));
+
+        nextButton.performClick();
+        index = myFragment.getCurrentDirection();
+        assertTrue(direction.get(index).equals("Down"));
+    }
+
+    @Test
+    public void ClickNextDirectionButton_ShowNextDirectionWhenNotAvailable_False(){
+        myFragment.addDirection("Up", "up");
+        myFragment.updateDirections();
+
+        int index = myFragment.getCurrentDirection();
+        ArrayList<String> direction = myFragment.getCompleteDirectionsList();
+
+        assertTrue(direction.get(index).equals("Up"));
+
+        nextButton.performClick();
+        int index1 = myFragment.getCurrentDirection();
+        assertEquals(index, index1);
+    }
+
+    @Test
+    public void ClickPreviousDirectionButton_ShowsPreviuousDirectionWhenAvailable_True(){
+        myFragment.addDirection("Up", "up");
+        myFragment.addDirection("Down", "down");
+        myFragment.updateDirections();
+        ArrayList<String> direction = myFragment.getCompleteDirectionsList();
+        nextButton.performClick();
+        int index = myFragment.getCurrentDirection();
+        assertTrue(index == 1);
+        previousButton.performClick();
+        index = myFragment.getCurrentDirection();
+        assertTrue(index == 0);
+    }
+
+    @Test
+    public void ClickPreviousDirectionButton_ShowPreviousDirectionWhenNotAvailable_False(){
+        myFragment.addDirection("Up", "up");
+        myFragment.updateDirections();
+        int index = myFragment.getCurrentDirection();
+        ArrayList<String> direction = myFragment.getCompleteDirectionsList();
+        previousButton.performClick();
+        int index1 = myFragment.getCurrentDirection();
+        assertEquals(index, index1);
+    }
+
+    @Test
+    public void AddDirection_AddsDirectionToList_True(){
+        myFragment.addDirection("Up", "up");
+        myFragment.updateDirections();
+        int index = myFragment.getCurrentDirection();
+        ArrayList<String> array = myFragment.getCompleteDirectionsList();
+        assertTrue(array.get(index).equals("Up"));
+
+        myFragment.addDirection("Left", "left");
+        myFragment.updateDirections();
+        assertTrue(array.get(index + 1).equals("Left"));
+    }
+
+    @Test
+    public void RemoveDirection_RemovesDirectionAtIndex_True(){
+        myFragment.addDirection("Up", "up");
+        myFragment.addDirection("Left", "left");
+        myFragment.updateDirections();
+
+        int index = myFragment.getCurrentDirection();
+        ArrayList<String> array = myFragment.getCompleteDirectionsList();
+        assertTrue(array.get(index).equals("Up"));
+
+        myFragment.removeDirection(0);
+        myFragment.updateDirections();
+
+        assertTrue(array.get(index).equals("Left"));
+    }
+
+    @Test
+    public void ClearDirections_SetsArrayToSize0_True(){
```

Comment:
Multiple methods are involved in this test updateDirections() and clearDirections(). Each should be tested individually